File: app/src/main/java/com/quran/labs/androidquran/common/LocalTranslation.java
package com.quran.labs.androidquran.common;

public class LocalTranslation {
  public final int id;
  public final String filename;
  public final String name;
  public final String translator;
  public final String translatorForeign;
  public final String url;
  public final int version;

  public LocalTranslation(int id, String filename, String name,
                          String translator, String translatorForeign, String url, int version) {
    this.id = id;
    this.filename = filename;
    this.name = name;
    this.translator = translator;
    this.translatorForeign = translatorForeign;
    this.url = url;
    this.version = version;
  }

  public String getTranslatorName() {
    final String result;
    if (this.translatorForeign != null) {
      result = this.translatorForeign;
    } else if (this.translator != null) {
      result = this.translator;
    } else {
      result = this.name;
    }
    return result;
  }
}

File: app/src/main/java/com/quran/labs/androidquran/common/LocalTranslation.kt
package com.quran.labs.androidquran.common

data class LocalTranslation(
  val id: Int = -1,
  val filename: String,
  val name: String = "",
  val translator: String? = "",
  val translatorForeign: String? = "",
  val url: String = "",
  val languageCode: String? = "",
  val version: Int = 1,
  val minimumVersion: Int = 2,
  val displayOrder: Int = -1) {

  fun getTranslatorName(): String {
    return when {
      !translatorForeign.isNullOrEmpty() -> translatorForeign
      !translator.isNullOrEmpty() -> translator
      name.isNotEmpty() -> name
      else -> filename
    }
  }
}
